packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Buttons/buttons.js

```diff
@@ -52,27 +63,84 @@ const Buttons = props => {
         right: 'flex-end'
     };
 
-    dynamicStyles.justifyContent = 'flex-start';
+    dynamicStyles.alignSelf = 'flex-start';
     if (textAlign) {
-        dynamicStyles.justifyContent = justifyMap[textAlign] || 'flex-start';
+        dynamicStyles.alignSelf = justifyMap[textAlign] || 'flex-start';
+        dynamicStyles.textAlign = textAlign;
     }
 
-    if (appearance === 'stacked') {
-        dynamicStyles.flexDirection = 'column';
+    let largestWidth = 0;
+    if (isSameWidth && appearance === 'inline') {
+        const totalChildren = Children.count(children);
+
+        Children.map(children, (child, index) => {
+            const newProps = child.props;
+
+            const allDone = new Promise(resolve => {
+                if (index + 1 === totalChildren) {
+                    resolve('done');
+                }
+            });
+
+            newProps.data.renderCallback = (width, changeWidth) => {
+                largestWidth = largestWidth > width ? largestWidth : width;
+                allDone.then(changeWidth(largestWidth));
```

Comment:
I still can't wrap my head around this even after talking through it with Dave :o Is it possible to just apply `flex-grow: 1` if we desire all the children to have same width?